### PR TITLE
Mnt ophyd fixes

### PR DESCRIPTION
**Pull request by tacaswell**
Changes found while trying to get ophyd + caproto to work.  I expect to push more here.

### CODE REVIEW

File: caproto/_commands.py [hunk 1]
```diff
@@ -563,10 +563,11 @@ class SearchRequest(Message):
...
     def __init__(self, name, cid, version):
         size, payload = padded_string_payload(name)
-        _len = len(name)
-        if _len > 40:
-            raise CaprotoValueError("EPICS imposes a 40-character limit on "
-                                    "strings. The " "string {!r} is {} "
+        rec, _, field = name.partition('.')
+        _len = len(rec)
+        if _len > 60:
+            raise CaprotoValueError("EPICS 3.14 imposes a 60-character limit "
```
Comment (danielballan) [inline caproto/_commands.py]: Have you encountered any documentation about how different versions of EPICS handle this? The version hardcoded here (3.14) may not match the actual `version` that we know here.

File: caproto/threading/client.py [hunk 2]
```diff
@@ -833,7 +833,12 @@ def get(self, *, count=None, as_string=False, as_numpy=True,
             return info['char_value']
         elif as_string and self.typefull in ca.enum_types:
             enum_strs = self.enum_strs
-            ret = [enum_strs[r] for r in info['value']]
+            ret = []
```
Comment (danielballan) [inline caproto/threading/client.py]: Can you add a comment on what we are silently ignoring here?

File: caproto/_commands.py (hunk 1 above)
Comment (tacaswell) [reply caproto/_commands.py]: See the link in the commit message http://www.aps.anl.gov/epics/tech-talk/2005/msg00083.php

Comment (klauer) [reply caproto/_commands.py]: I think we can safely ignore anything below 3.14.

File: caproto/threading/client.py (hunk 2 above)
Comment (klauer) [reply caproto/threading/client.py]: It's not impossible to have invalid enum values that don't correspond to any string (I think that's what @tacaswell ran into here)

**Comment by klauer** [comment]
Is this still not ready @tacaswell?

File: caproto/_commands.py (hunk 1 above)
Comment (danielballan) [reply caproto/_commands.py]: Fair enough. I like the idea that if there are any Reagan-era Star Wars satellites out there, we could talk to them with Python 3.6. But until we find one to test against.... ;-)

**Comment by tacaswell** [comment]
I have put 0 cycles into this since pushing.  This is just to record the hacking I did before giving up on the effort.

**Comment by danielballan** [comment]
Rebased to get the test fixes in #115

**Comment by danielballan** [comment]
There are more fixes to come until ophyd + caproto passes the ophyd + pyepics tests, but we might as well merge them incrementally since the work has stalled.

**Pull request merged**